File: client-scalaz/src/main/scala/org/openapitools/client/api/PurchaseItemListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PurchaseItemListResponse._

case class PurchaseItemListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[PurchaseItemResponse]],
count: Option[Integer],
hasMoreResults: Option[Boolean],
start: Option[Integer],
limit: Option[Integer],
countTotal: Option[Long],
returning: Option[String])

object PurchaseItemListResponse {
  import DateTimeCodecs._

  implicit val PurchaseItemListResponseCodecJson: CodecJson[PurchaseItemListResponse] = CodecJson.derive[PurchaseItemListResponse]
  implicit val PurchaseItemListResponseDecoder: EntityDecoder[PurchaseItemListResponse] = jsonOf[PurchaseItemListResponse]
  implicit val PurchaseItemListResponseEncoder: EntityEncoder[PurchaseItemListResponse] = jsonEncoderOf[PurchaseItemListResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/purchase_item_list_response.clj
(ns sirqul-io-t-platform.specs.purchase-item-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.purchase-item-response :refer :all]
            )
  (:import (java.io File)))


(def purchase-item-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of purchase-item-response-spec)
   (ds/opt :count) int?
   (ds/opt :hasMoreResults) boolean?
   (ds/opt :start) int?
   (ds/opt :limit) int?
   (ds/opt :countTotal) int?
   (ds/opt :returning) string?
   })

(def purchase-item-list-response-spec
  (ds/spec
    {:name ::purchase-item-list-response
     :spec purchase-item-list-response-data}))
